website/database.py

```diff
@@ -608,6 +611,13 @@ def batch_get_adventures(self, adventure_ids):
         keys = {id: {"id": id} for id in adventure_ids}
         return ADVENTURES.batch_get(keys) if keys else {}
 
+    def get_public_adventures_filters(self, key):
+        return PUBLIC_ADVENTURES_FILTERS.get_all(key)
```

Comment:
The levels should never be filtered out alone, but with combination with the language/tags. I do think if you select level 1, you'd need to see languages/tags for that level, right? 

What you're saying is that based on the selected language, we show available levels/tags?

Reply:
Also, about scanning the table. This will have to be triggered whenever the user refreshes the page or applies some filters. I don't think it'd be heavey, but let's see its performance once it's done!

Reply:
I know, scanning a table is usually bad. But in this case, the table isn't too big I don't think.

> I do think if you select level 1, you'd need to see languages/tags for that level, right?

There's no arbitrary navigation hierarchy here though; there is a definite order. No real user is going to pick the level first, and only then see what language they can choose. They will begin with picking a language, because it will be *their* language.

It does make sense to filter the list of levels down when a language is picked, and perhaps the list of tags as well.

But I don't think it needs to be more complicated than that, and selecting language, then level, and having the list of tags filtered down to only that combination of language and level is possible, but overkill. So let's not design for it yet.